agent.go

```diff
@@ -665,6 +665,12 @@ func (ep *endpoint) addServiceInfoToCluster(sb *sandbox) error {
 }
 
 func (ep *endpoint) deleteServiceInfoFromCluster(sb *sandbox, method string) error {
+
+	if !ep.serviceEnabled {
```

Comment:
The possibility of concurrent calls too deleteServiceInfoFromCluster is from sb.DisableService (which calls getConnectedEndpoints) and ep.Leave(). The latter fetches the endpoint from the store. So the `serviceEnabled` field has to be marshalled. Also, after setting it ep has to be updated in the store.